aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1081 @@
+        let len = vector::length(&owners_to_remove);
+        let owners = &mut multisig_account_resource.owners;
+        let owners_removed = vector::empty<address>();
+        while (i < len) {
+            let owner_to_remove = *vector::borrow(&owners_to_remove, i);
+            let (found, index) = vector::index_of(owners, &owner_to_remove);
+            // Only remove an owner if they're present in the owners list.
+            if (found) {
+                vector::push_back(&mut owners_removed, owner_to_remove);
+                vector::swap_remove(owners, index);
+            };
+            i = i + 1;
+        };
+
+        // Make sure there's still at least as many owners as the number of signatures required.
+        // This also ensures that there's at least one owner left as signature threshold must be > 0.
+        assert!(
+            vector::length(owners) >= multisig_account_resource.signatures_required,
+            error::invalid_state(ENOT_ENOUGH_OWNERS),
+        );
+
+        emit_event(&mut multisig_account_resource.remove_owners_events, RemoveOwnersEvent { owners_removed });
+    }
+
+    /// Update the number of signatures required to execute transaction in the specified multisig account.
+    /// This can only be invoked by the multisig account itself, through the proposal flow.
+    public entry fun update_signatures_required(
+        multisig_account: &signer, new_signatures_required: u64) acquires MultisigAccount {
+        let multisig_address = address_of(multisig_account);
+        assert_multisig_account_exists(multisig_address);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_address);
+        // Short-circuit if the new number of signatures required is the same as before.
+        // This avoids emitting an event.
+        if (multisig_account_resource.signatures_required == new_signatures_required) {
+            return
+        };
+        let num_owners = vector::length(&multisig_account_resource.owners);
+        assert!(
+            new_signatures_required > 0 && new_signatures_required <= num_owners,
+            error::invalid_argument(EINVALID_SIGNATURES_REQUIRED),
+        );
+
+        let old_signatures_required = multisig_account_resource.signatures_required;
+        multisig_account_resource.signatures_required = new_signatures_required;
+        emit_event(
+            &mut multisig_account_resource.update_signature_required_events,
+            UpdateSignaturesRequiredEvent {
+                old_signatures_required,
+                new_signatures_required,
+            }
+        );
+    }
+
+    /// Create a multisig transaction, which will have one approval initially (from the creator).
+    ///
+    /// @param target_function The target function to call such as 0x123::module_to_call::function_to_call.
+    /// @param args Vector of BCS-encoded argument values to invoke the target function with.
+    public entry fun create_transaction(
```

Comment:
If an owner's private key is leaked and create transactions at a high rate, will that block the wallet?

Reply:
Yes. But this is the case in general for in-order transaction execution anyway. The broader question would be whether we want to enforce in-order transactions, which is generally the case for most popular multisig account products on other chains such as Gnosis Safe and the rule for normal account transactions on Aptos.